lib/common/compiler.h

```diff
@@ -92,7 +92,7 @@
 #if defined(_MSC_VER) && (defined(_M_X64) || defined(_M_I86))  /* _mm_prefetch() is not defined outside of x86/x64 */
 #  include <mmintrin.h>   /* https://msdn.microsoft.com/fr-fr/library/84szxsww(v=vs.90).aspx */
 #  define PREFETCH(ptr)   _mm_prefetch((const char*)ptr, _MM_HINT_T0)
-#elif defined(__GNUC__)
+#elif defined(__GNUC__) && ((__GNUC__ >= 4) || (__GNUC__ == 3) && (__GNUC_MINOR__ >= 1))
```

Comment:
Would it be better to test against 3.0.3 with this change?
```diff
-#elif defined(__GNUC__) && ((__GNUC__ >= 4) || (__GNUC__ == 3) && (__GNUC_MINOR__ >= 1))
+#elif defined(__GNUC__) && (__GNUC__ >= 4 || __GNUC__ == 3 && (__GNUC_MINOR__ >= 1 || __GNUC_MINOR__ == 0 && defined(__GNUC_PATCHLEVEL__) && __GNUC_PATCHLEVEL__ >= 3))
```
BTW, I dropped some unneeded parenthesis but I need to check this.

Hmm, after all, shall I just change the commit message to `__builtin_prefetch did probably not exist before gcc 3.0.3.` and keep a simple test for `3.1` instead of `3.0.3`?

Reply:
The PR title is fine.
Stick to 3.1, and do not rely on operator priority to interpret the version test.
Parenthesis are fine. They help ensure the test performed is the wanted one.

Reply:
I'm so sorry, I followed the link for gcc 3.1 link thinking I had followed the one for gcc 3.0.3!!!
(So when I found `builtin_prefetch` I wrongly though it was available in 3.0.3. :(

So everything is already OK. The commit title is OK as well as test.
Hmm, should I drop some parenthesis in the test? I guess this should be OK:
```
+#elif defined(__GNUC__) && (__GNUC__ >= 4 || __GNUC__ == 3 && __GNUC_MINOR__ >= 1)
```

Reply:
no you should ___add___ parenthesis in the test, anywhere there could be a doubt on operator priority.
At the very least : 
```
#elif defined(__GNUC__) && (__GNUC__ >= 4 || (__GNUC__ == 3 && __GNUC_MINOR__ >= 1))
```

Reply:
Yep, but some purists like to live dangerously, putting parenthesis only where they are strictly required.
I tend to choose the safe side, but maybe we should replace
```
#elif defined(__GNUC__) && ((__GNUC__ >= 4) || ( (__GNUC__ == 3) && (__GNUC_MINOR__ >= 1) ) )
```
by
```
#elif defined(__GNUC__) && (__GNUC__ >= 4 || (__GNUC__ == 3 && __GNUC_MINOR__ >= 1))
```
What would you recommend?

Reply:
either version is good to me

Reply:
Here also, I'm fine with whatever works best for HaikuPorts.
Squashing is slightly better, but I don't mind if it's not ,
so if there is a reason to prefer the non-squashed version, I'm fine with it and can merge it.

Reply:
I also prefer the squashed version. It's there in a minute.

Reply:
I've added a space after one of the opening parenthesis, otherwise it would break the symetry with the space before it"s corresponding closing parenthesis ;)